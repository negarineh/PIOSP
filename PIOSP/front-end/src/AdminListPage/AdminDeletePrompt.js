import React from "react"; 
import PropTypes from 'prop-types';

import { Button, Modal} from 'semantic-ui-react';

// User delete component
export default class AdminDeletePrompt extends React.Component {

  // render 
  render() {
    const {show, user, hideDelete, userDelete} = this.props;
    return (
      
      <Modal size="small" open={show} onClose={hideDelete} 
             style={{ position: 'absolute',
                      top: '50%',
                      left: '50%',
                      transform: 'translate(-50%, -50%)',
                      textAlign:'justify'}}>
          <Modal.Header> 
            Delete Account
          </Modal.Header>
          <Modal.Content style={{ width: '80%'}}>
            <p style={{ padding: '15px', textAlign:'justify'}}>Are you sure you want to delete <strong>{user.name}</strong> account? </p>
          </Modal.Content>
          <Modal.Actions>
            <Button negative onClick={hideDelete}> 
              No
            </Button>
            <Button style={{ marginLeft: '74%', height:'37px'}} positive icon='checkmark' labelPosition='right' content='Yes' onClick={userDelete}/>
          </Modal.Actions>
        </Modal>
    );
  }
}


AdminDeletePrompt.defaultProps = {
  show: false,
  user: {},
  hideDelete: () => {},
  userDelete: () => {}
}
// prop checks
AdminDeletePrompt.propTypes = {
  show: PropTypes.bool.isRequired,
  user: PropTypes.object.isRequired,
  hideDelete: PropTypes.func.isRequired,
  userDelete: PropTypes.func.isRequired,
}
