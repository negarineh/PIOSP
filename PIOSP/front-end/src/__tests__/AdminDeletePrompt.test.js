import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import AdminDeletePrompt from "../AdminListPage/AdminDeletePrompt";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('UserDeletePrompt component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {show: true, user: {}, hideDelete: ()=>{}, userDelete: ()=>{}};
      const wrapper = shallow(<AdminDeletePrompt {...props}/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
