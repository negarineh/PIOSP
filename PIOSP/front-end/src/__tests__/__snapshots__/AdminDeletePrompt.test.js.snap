// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDeletePrompt component render() should render the component 1`] = `
<Modal
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  open={true}
  size="small"
  style={
    Object {
      "left": "50%",
      "position": "absolute",
      "textAlign": "justify",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
    }
  }
>
  <ModalHeader>
    Delete Account
  </ModalHeader>
  <ModalContent
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <p
      style={
        Object {
          "padding": "15px",
          "textAlign": "justify",
        }
      }
    >
      Are you sure you want to delete 
      <strong />
       account? 
    </p>
  </ModalContent>
  <ModalActions>
    <Button
      as="button"
      negative={true}
      onClick={[Function]}
    >
      No
    </Button>
    <Button
      as="button"
      content="Yes"
      icon="checkmark"
      labelPosition="right"
      onClick={[Function]}
      positive={true}
      style={
        Object {
          "height": "37px",
          "marginLeft": "74%",
        }
      }
    />
  </ModalActions>
</Modal>
`;
